framework/platform/bpl/uci/bpl_uci.cpp

```diff
@@ -0,0 +1,263 @@
+    {
+        struct uci_option *opt = uci_to_option(elm);
+        // Only UCI_TYPE_STRING is supported
+        if (o->type == UCI_TYPE_STRING) {
+            params.emplace(o->e.name, o->v.string);
+        }
+    }
+
+    return true;
+}
+
+bool uci_get_entry_type(const std::string &config_file, const std::string &entry_name,
+                        std::string &entry_type)
+{
+    //entry: config_file: entry_name
+    LOG(TRACE) << "entry: " << config_file << ": " << entry_name;
+
+    struct uci_context *ctx = nullptr;
+    if (!(ctx = uci_alloc_context())) {
+        LOG(ERROR) << "uci context allocate failed!";
+        return false;
+    }
+
+    struct uci_ptr ptr = nullptr;
+    auto &path         = std::format("{}.{}", config_file, entry_name);
+    if (uci_lookup_ptr(cont, &ptr, path.c_str(), true) != UCI_OK || !ptr.s) {
+        LOG(ERROR) << "uci lookup package failed!";
+        uci_free_context(ctx);
+        return false;
+    }
+
+    entry_type = ptr->s.type;
+    uci_free_context(ctx);
+    return false;
+}
+
+bool uci_get_param(const std::string &config_file, const std::string &entry_type,
+                   const std::string &entry_name, const std::string &param_name, std::string &param)
+{
+    //entry: config_file: entry_type(entry_name), param
+    LOG(TRACE) << "entry: " << config_file << ": " << entry_type << "(" << entry_name << ")"
+               << ", " << param_name;
+
+    struct uci_context *ctx = nullptr;
+    if (!(ctx = uci_alloc_context())) {
+        LOG(ERROR) << "uci context allocate failed!";
+        return false;
+    }
+
+    struct uci_ptr ptr = nullptr;
+    auto &path         = std::format("{}.{}.{}", config_file, entry_name, param_name);
+    if (uci_lookup_ptr(cont, &ptr, path.c_str(), false) != UCI_OK || !ptr.o) {
+        LOG(ERROR) << "uci lookup package failed!";
+        uci_free_context(ctx);
+        return false;
+    }
+
+    if (ptr.o->type != UCI_TYPE_STRING) {
+        LOG(ERROR) << "list values are not supported!";
+        uci_free_context(ctx);
+        return false;
+    }
+
+    param = ptr.o->v.string;
+    uci_free_context(ctx);
+    return true;
+}
+
+bool uci_delete_entry(const std::string &config_file, const std::string &entry_type,
+                      const std::string &entry_name)
+{
+    //entry: config_file: entry_type(entry_name)
+    LOG(TRACE) << "entry: " << config_file << ": " << entry_type << "(" << entry_name << ")";
+
+    struct uci_context *ctx = nullptr;
+    if (!(ctx = uci_alloc_context())) {
+        LOG(ERROR) << "uci context allocate failed!";
+        return false;
+    }
+
+    struct uci_ptr ptr = nullptr;
+    auto &path         = std::format("{}.{}.{}", config_file, entry_name, param_name);
+    if (uci_lookup_ptr(ctx, &ptr, path.c_str(), false) != UCI_OK || !ptr.s) {
+        LOG(ERROR) << "uci lookup package failed!";
+        uci_free_context(ctx);
+        return false;
+    }
+
+    uci_delete(ctx, ptr);
+
```

Comment:
Missing `uci_free_context(ctx);` in a lot of functions.
See my `shared_ptr` comment to remove this dependency at all.

Reply:
Added `shared_ptr` for the context